### PR TITLE
CLI binary improvements

### PR DESCRIPTION
**Pull request by str4d**
Cherry-picked from the following upstream PRs:

- bitcoin/bitcoin#5936
- bitcoin/bitcoin#7550
- bitcoin/bitcoin#7989
- bitcoin/bitcoin#7957
- bitcoin/bitcoin#9067
- bitcoin/bitcoin#9220

Excludes any changes that affected the QT code.

### CODE REVIEW

**Comment by str4d** [comment]
Rebased on master to fix merge conflict.

File: src/bitcoin-tx.cpp [hunk 1]
```diff
@@ -201,13 +201,18 @@ static void MutateTxAddInput(CMutableTransaction& tx, const string& strInput)
...
-    string strVout = strInput.substr(pos + 1, string::npos);
+    string strVout = vStrInputParts[1];
     int vout = atoi(strVout);
     if ((vout < 0) || (vout > (int)maxVout))
         throw runtime_error("invalid TX input vout");
 
+    // extract the optional sequence number
+    uint32_t nSequenceIn=std::numeric_limits<unsigned int>::max();
+    if (vStrInputParts.size() > 2)
+        nSequenceIn = atoi(vStrInputParts[2]);
```
Comment (daira) [inline src/bitcoin-tx.cpp]: Another case of atoi being used on untrusted input; see #2075.

File: src/bitcoind.cpp [hunk 2]
```diff
@@ -88,7 +88,7 @@ bool AppInit(int argc, char* argv[])
         }
 
         fprintf(stdout, "%s", strUsage.c_str());
-        return false;
+        return true;
     }
```
[the remark this answers is not in the document]
Comment (daira) [reply src/bitcoind.cpp]: Oh, looks like that got changed to ``exit(EXIT_FAILURE)``. But I still think it should probably be ``return false``, for consistency with other returns in that function.

[the remark this answers is not in the document]
Comment (str4d) [reply src/bitcoind.cpp]: This will be addressed when we pull in bitcoin/bitcoin#10447 and bitcoin/bitcoin#11511 (which we can do in a subsequent PR).

**Comment by str4d** [comment]
Rebased on master to fix merge conflict, and address some of the non-blocking comments.

@zkbot r+

**Comment by str4d** [comment]
`util-test` failed because I didn't tweak the test case data for Zcash (binary renaming, and re-prefixing t-addrs). Pushed a trivial commit to fix.

**Comment by str4d** [comment]
@zkbot force

**Comment by str4d** [comment]
@zkbot r+ ee6220c0ed9ef90ae9317231463099c02c303bca

**Pull request merged**